addons/base_import_module/models/ir_module.py

```diff
@@ -218,6 +218,18 @@ def _import_module(self, module, path, force=False, with_demo=False):
                 _logger.info('module %s: no translation for language %s', module, lang)
         translation_importer.save(overwrite=True)
 
+        article = f"{module}.welcome_article_body"
+        article_body = self.env.ref(article, raise_if_not_found=False)
+        article_record = self.env.ref(f"{module}.welcome_article", raise_if_not_found=False)
+        if article_body and article_record:
+            user = self.env.user
+            render_ctx = {'object': user}
+            body = self.env['ir.qweb']._render(article, render_ctx,
+                                            raise_if_not_found=False,
+                                            lang=self.env.user.lang)
```

Comment:
I don't understand why only update body for user.lang

Could you also rebsase the code? odoo/odoo#173710 can translate for industry also

Reply:
Knowledge articlee bodies are not translatable, we just want to set it in the language of the user that installed the module.

It is how it is done for the onboarding knowledge article in standard: https://github.com/odoo/enterprise/blob/17.0/knowledge/models/res_users.py#L18

Reply:
I am not familiar with the business logics. But if it is for knowledge, why not override the method in knowledge? Is it necessary even if the knowledge is not installed?

Reply:
it when you install an industry which is basically a imported module. It has nothing to do with knwoledge, but with ow the industries are workging (https://github.com/odoo/industry)

Reply:
OK, so the `knowlege article` is not for the knowledge module, right?

Reply:
the knowledge article is a data that will be in each industry module( importable modules). Those industry modules depends on knowledge, and so knowledge is installed when the module is imported.

Reply:
I am not sure if it can be treated as translations even if the body field is not translated. If so, you can override _load_module_terms, and you can tell if it is for industry module by parameter imported_module
https://github.com/odoo/odoo/blob/18e13245e052efd4ddbcec7bc21ac6f84d548b89/addons/base_import_module/models/ir_module.py#L209

Reply:
Hi @HydrionBurst 
The goal here is to have the knowledge article in the language of the admin. Since `body` of knowledge.article is not translatable, we need to render it in the language selected by the user at the time of the installation. 
This prevents us to do it in the `_load_module_terms`, because it will be done in a single language (whereas `_load_module_terms` loops over all installed languages). I know it is not ideal if the user has several languages installed (I thought about creating one article per language) but this article aims at guiding the firsts steps of the admin only. And with the new option coming up to create a trial with an industry module directly, there will be only one language so this issue is reduced.
I don't really see a better option for now, but if you do, don't hesitate to share.